pdc_client/__init__.py

```diff
@@ -89,15 +95,104 @@ def read_config_file(server_alias):
     return result
 
 
+class MultiPageBeanBag(BeanBag):
+    """
+    Add this class to support to return all pages of data by default in a get request.
+
+    This class is based on the patch:
+    https://github.com/lkocman/beanbag/commit/407fff011c9e1f3883734f44a3d48fe041b2323
+    9#diff-5ad74912650ff9cf4a6754f8176f68faR336
+
+    We can use the class directly if the patch adds in the master of beanbag in the futrue.
+    """
+
+    def is_multi_page_response(self, response):
+        """
+        Check if response is a dict or not
+        """
+        if isinstance(response, dict):
+            if set([u'count', u'next', u'results', u'previous']).issubset(set(response.keys())):
+                return True
+        return False
+
+    def url2path_ext(self, url):
+        """
+        @param url
+        returns path, params
+        """
+        parsed = urlparse(url)
+        return parse_qs(parsed.query)
+
+    def _make_request(self, path, verb, params, body):
+        """
+        Rename the make_request of BeanBag to _make_request.
+        As the class type is MultiPageBeanBagBase, we can't
+        use a super function.
```

Comment:
I'm not sure I understand this comment. `MultiPageBeanBagBase` is a subclass of `BeanBag`, so using `super` would call the method on parent, right?

Reply:
In BeanBag class if you create a subclass, the base class will change to <classname>+Base, I saw this is inside beanbag __new__ function. And a super function will return error as the base class is not BeanBag.
So we just rename the make_request like what original patch does, not call super.make_request in the new function.
That's the comment's meaning.